rmmxapianapi.cc

```diff
@@ -523,6 +524,58 @@ extern "C" {
       return numfolders;
     }
 
+    // returns a pair: [total, unread] in `results[]`
+    int EMSCRIPTEN_KEEPALIVE getFolderMessageCounts(const char *folderName, int results[]) {
+        if (!dbc) return 0;
+
+        Xapian::QueryParser queryparser;
+        queryparser.set_database(dbc->db);
+        queryparser.add_boolean_prefix("folder", "XFOLDER:");
+        queryparser.add_boolean_prefix("flag", "XF");
+
+        string queryString = "folder:\"";
+        queryString += folderName;
+        queryString += "\"";
+
+        try {
+            {
+                Xapian::Enquire enquire(dbc->db);
+                Xapian::Query query = queryparser.parse_query(
+                    queryString + " AND NOT flag:seen",
+                    Xapian::QueryParser::FLAG_DEFAULT | Xapian::QueryParser::FLAG_PARTIAL
+                );
+
+                enquire.set_query(query);
+                Xapian::MSet mset = enquire.get_mset(0, UINT_MAX);
```

Comment:
This results in Xapian having to return a result set containing every single document matching the query, but you only want to know the number of entries there would be in that set, not what any of them actually are.

You can tell Xapian you are only interested in the number of results by specifying `0` for `maxitems` and then a large number for `checkatleast` instead:

```
Xapian::MSet mset = enquire.get_mset(0, 0, UINT_MAX);
```

Reply:
That gives me a `.size()` of 0, but the numbers from `.get_matches_estimated()` seem to me matching reality (I don't have a big enough static dataset to verify this on, unfortunately) – can I reasonably assume that with a big enough `checkatlest` the estimated numbers will be equal to the current state of things?

Reply:
Oh sorry - yes, you also need to look at `.get_matches_estimated()` instead, and the estimate should be exact in this case since we're forcing the matcher to consider all possibly matching documents.